### PR TITLE
Apply result size limits

### LINKED ISSUE #16: Consider limiting the size of a single check result (total bytes and number of keys)
The size of check results is currently unbounded, this leads to problems where users generate (mostly accidentally) Megabytes (!) of result data for a single check. As the data is stored in JSON format in Redis (and additionally in KairosDB), we might run into memory issues (e.g. Redis memory fragmentation and total database size).

A simple and effective approach would be to introduce a reasonable (configurable) maximum for check results, let's say 64KiB.

### PR DESCRIPTION
**Pull request by mohabusama**
Current default limits are:

- Size: 64KB
- Keys: 1000

Fixes #16

### CODE REVIEW

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 1]
```diff
@@ -1042,6 +1051,25 @@ def _store_check_result(self, req, result):
...
 
+    def _check_result_limit(self, result):
+        if isinstance(result, dict):
+            key_count = len(flatten(result).keys())
+            if key_count > self.max_result_keys:
+                raise ResultSizeError(
+                    'Result keys count ({}) exceeded the maximum value: {}'.format(key_count, self.max_result_keys))
+
+        result_str = ''
+        if isinstance(result, dict) or isinstance(result, list):
```
Comment (hjacobs) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: let's keep it simple and use `json.dumps(...)` always (one less branch to maintain)

**Comment by hjacobs** [comment]
@mohabusama can you analyze our current check results and find out what would break and how to find reasonable values? (you can use the controller's REST API for this)

File: zmon_worker_monitor/zmon_worker/tasks/main.py (hunk 1 above)
Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: Done.

File: zmon_worker_monitor/zmon_worker/tasks/main.py [hunk 2]
```diff
@@ -1058,11 +1058,7 @@ def _check_result_limit(self, result):
                 raise ResultSizeError(
                     'Result keys count ({}) exceeded the maximum value: {}'.format(key_count, self.max_result_keys))
 
-        result_str = ''
-        if isinstance(result, dict) or isinstance(result, list):
-            result_str = json.dumps(result).replace(' ', '')
-        elif isinstance(result, basestring):
-            result_str = result
+        result_str = json.dumps(result).replace(' ', '')
```
Comment (hjacobs) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: why do we replace spaces? maybe just configure appropriate `json.dumps` separators? https://docs.python.org/3/library/json.html#json.dump

Comment (mohabusama) [inline zmon_worker_monitor/zmon_worker/tasks/main.py]: Just to minimize the result size before the check.
Updated to use `separators`

**Comment by hjacobs** [comment]
@mohabusama can you resolve the conflicts?

**Comment by hjacobs** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**